test/functional/feature_config_args.py

```diff
@@ -319,6 +321,25 @@ def test_connect_with_seednode(self):
                     unexpected_msgs=seednode_ignored):
                 self.restart_node(0, extra_args=[connect_arg, '-seednode=fakeaddress2'])
 
+            # Make sure -noconnect soft-disables -listen and -dnsseed.
+            # Need to temporarily remove these settings from the config file in
+            # order for the two log messages to appear
+            self.nodes[0].replace_in_config([("bind=", "#bind="), ("dnsseed=", "#dnsseed=")])
+            with self.nodes[0].assert_debug_log(expected_msgs=addcon_thread_started) as info:
+                self.restart_node(0, extra_args=[connect_arg])
+            self.nodes[0].replace_in_config([("#bind=", "bind="), ("#dnsseed=", "dnsseed=")])
+            for msg in [dnsseed_disabled, listen_disabled]:
+                if msg not in info.log:
+                    raise AssertionError(f"Expected {msg!r} in -noconnect log message")
+
+            # Make sure -proxy and -noconnect warn about -dnsseed setting being
+            # ignored, just like -proxy and -connect do.
+            with self.nodes[0].assert_debug_log(expected_msgs=addcon_thread_started) as info:
+                self.restart_node(0, extra_args=[connect_arg, '-dnsseed', '-proxy=localhost:1080'])
+            for msg in dnsseed_ignored:
+                if msg not in info.log:
+                    raise AssertionError(f"Expected {msg!r} in -noconnect log message")
+
```

Comment:
nit :  not sure about this but I think it would be useful to check that when -noconnect/connect=0 is set no outbound connections are made.

Reply:
That would be nice and if I can figure out a way to check for that more directly I could drop the second half of this new test. It looks like there is a log message "%i block-relay-only anchors will be tried for connections" that gets printed when `m_use_addrman_outgoing` is false so maybe I could check for that when -noconnect is specified, and have better confirmation that noconnect is working correctly, than this `dnsseed_ignored` check provides.

I think the first half of the test checking for `dnsseed_disabled` and `listen_disabled` is useful in either case since it is providing test coverage for InitParameterInteraction code handling noconnect instead of AppInitMain code handling it.